mozilla/fxa
# feat(react): Build SigninUnblock page in React

## Because

* We are converting backbone signin routes to react

## This pull request

* Creates React page for SigninUnblock, including container component, unit tests, stories and l10n
* Updates Signin page to redirect to SigninUnblock when requests are throttled or blocked
* Conditionally renders React page if feature flag is on
* Update error handling in Signin and Signup to support throttling errors with localized retry after values
## Issue that this pull request solves

Closes: #FXA-9030

## Checklist

_Put an `x` in the boxes that apply_


## Screenshots (Optional)

Backbone (some styling appears to be broken):

![image](https://github.com/mozilla/fxa/assets/22231637/eb026a16-b056-4916-ad86-407838201989)

React:

![image](https://github.com/mozilla/fxa/assets/22231637/212c1cec-a563-4acd-a6ad-5da9b321b2b3)

## Other information (Optional)

To test locally:
1. Enable customs temporarily by settings `customsUrl` in `packages/fxa-auth-server/config/dev.json` to `http://localhost:7000`
2. Create a new account.
3. Sign out.
4. Go to [signin with React](http://localhost:3030/?forceExperiment=generalizedReactApp&forceExperimentGroup=react)
5. Enter the email used to create the account
6. Enter the wrong password and submit 3x times (observe tooltip error), enter the correct password and submit
7. Expect to be redirected to "Authorize this sign-in", code will be in inbox (check with inbox logs or maildev)

Test the flow when entering the correct or incorrect unblock code, entering an incorrect password 4 times instead or 3, resending the code and trying the new code.

Will need to create multiple accounts for testing - after successfully going through the unblock flow, a second attempt will be throttled with no option to unblock.

Additionally, this PR includes a fix for localized retry value showing up as a variable in react signin/signup flows. To test, set the preferred browser language to a supported language other than English. When the throttling error is shown in a a banner, the retry value (e.g., "in 15 minutes") should be localized.

NOT INCLUDED YET in this PR:
- container tests for signin_unblock
- noticed a localization issue on the backbone index page for throttled errors - retry after value is localized but not string.
